Assets/Scripts/InventoryScripts/Pickupable.cs

```diff
@@ -92,30 +132,33 @@ public void CloseItemDescription()
     /// <param name="player">The player interacting</param>
     public void Interact(GameObject player)
     {
-        _playerController.LockCharacter(true);
-        _playerInteractBehavior.StopDetectingInteractions();
-        _tabbedMenu.DisplayDialogue("", _itemDescription.Description);
-        _tabbedMenu.ToggleDialogue(true);
-        _tabbedMenu.ClearDialogueOptions();
-        _tabbedMenu.DisplayDialogueOption(_itemDescription.ExitResponse, click: () => { CloseItemDescription(); });
+        if(_isInteractive)
+        {
+            _playerController.LockCharacter(true);
+            _playerInteractBehavior.StopDetectingInteractions();
+            _tabbedMenu.DisplayDialogue("", _itemDescription.Description);
+            _tabbedMenu.ToggleDialogue(true);
+            _tabbedMenu.ClearDialogueOptions();
+            _tabbedMenu.DisplayDialogueOption(_itemDescription.ExitResponse, click: () => { CloseItemDescription(); });
 
-        InventoryHolder inventoryHolder = player.GetComponent<InventoryHolder>();
+            InventoryHolder inventoryHolder = player.GetComponent<InventoryHolder>();
 
-        //if it is not already in inventory
-        if (inventoryHolder != null && !inventoryHolder.InventorySystem.ContainsItem(myData, out _))
-        {
-            //Debug.Log("Got item");
-            inventoryHolder.InventorySystem.AddToInventory(myData, 1, out _);
-            Destroy(gameObject);
-        }
-        //if it is already in inventory (prevents doubles in inventory)
-        else if(inventoryHolder != null)
-        {
-            Destroy(gameObject);
-        }
-        else
-        {
-            Debug.LogError("InventoryHolder component not found on player.");
+            //if it is not already in inventory
+            if (inventoryHolder != null && !inventoryHolder.InventorySystem.ContainsItem(myData, out _))
```

Comment:
This can be cleaned up by switching the conditional to 
`if(inventoryholder == null) { Debug.LogError(""); return;}` then have the `ContainsItem` condition. This is generally just easier to read and good practice too (in my experience) [early return](https://medium.com/swlh/return-early-pattern-3d18a41bba8)
 wherever possible instead of chain or nested conditionals.